src/toit.cc

```diff
@@ -15,375 +15,51 @@
-  printf("  [-h] [--help]                        // This help message.\n");
-  printf("  [--version]                          // Prints version information.\n");
-  printf("  [-O<level>]                          // Set optimization level (default = 1).\n");
-  printf("  [-X<flag>]*                          // Provide a compiler flag.\n");
-  printf("  [-b <snapshot>]                      // Use a specific boot snapshot, default is the adjacent toit.run.snapshot.\n");
-  printf("  [--dependency-file <file>]           // Write a dependency file ('-' for stdout).\n");
-  printf("  [--dependency-format {plain|ninja}]  // The format of the dependency file.\n");
-  printf("  [--project-root <path>]              // Path to the project root. Any package.lock file must be in that folder.\n");
-  printf("  [--force]                            // Finish compilation even with errors (if possible).\n");
-  printf("  [-Werror]                            // Treat warnings like errors.\n");
-  printf("  [--show-package-warnings]            // Show warnings from packages.\n");
-  printf("  { <snapshot> <args>... |             // Run snapshot file.\n");
-  printf("    <toitfile> <args>... |             // Run Toit file.\n");
-  printf("    -w <snapshot> <toitfile> |         // Write snapshot file.\n");
-  printf("    -s <expression> |                  // Evaluate Toit expression.\n");
-  printf("    --analyze <toitfiles>...           // Analyze Toit files.\n");
-  printf("  }\n");
-  exit(exit_code);
-}
-
-// Prints the version and exits.
-static void print_version() {
-  printf("Toit version: %s\n", vm_git_version());
-  exit(0);
-}
-
 int main(int argc, char **argv) {
-  Flags::process_args(&argc, argv);
-  if (argc < 2) print_usage(1);
-
   FlashRegistry::set_up();
   OS::set_up();
   ObjectMemory::set_up();
 
   int exit_state = 0;
-  char* boot_bundle_path = null;
-  if (strcmp(argv[1], "-b") == 0) {
-    // The wrapping boot bundle is passed after the '-b' arg.
-    if (argc < 3) {
-      fprintf(stderr, "Missing argument to '-b' flag\n");
-      print_usage(1);
-    }
-    // Malloc the path to ensure freeing works consistently.
-    boot_bundle_path = unvoid_cast<char*>(malloc(strlen(argv[2]) + 1));
-    strcpy(boot_bundle_path, argv[2]);
-    argc -= 2;
-    argv += 2;
-  }
-
-  // Help must be used on its own.
-  if (strcmp(argv[1], "-h") == 0 || strcmp(argv[1], "--help") == 0) {
-    if (argc != 2) {
-      fprintf(stderr, "Can't have options with '%s'\n", argv[1]);
-      print_usage(1);
-    }
-    print_usage(0);
-  }
-
-  // Version must be used on its own.
-  if (strcmp(argv[1], "--version") == 0) {
-    if (argc != 2) {
-      fprintf(stderr, "Can't have options with '%s'\n", argv[1]);
-      print_usage(1);
-    }
-    print_version();
-  }
-
-  // TODO(2663): remove support for '-r'.
-  if (strcmp(argv[1], "-r") == 0 ||
-      SnapshotBundle::is_bundle_file(argv[1])) {
+  if (argc >= 2 && SnapshotBundle::is_bundle_file(argv[1])) {
```

Comment:
Do we need all this handling?

Reply:
It's to support fast "run with a snapshot".
For all other cases we need to invoke the compiler anyway, so it feels like running a Toit program first doesn't hurt as much.
We can add more fast cases, if we feel like going through the Toit code adds too much overhead.